### PR TITLE
feat(ottl): add Bool function

### LINKED ISSUE #44770: [pkg/ottl] Support string to boolean conversion.
### Component(s)

pkg/ottl

### Is your feature request related to a problem? Please describe.

Original discussion in https://cloud-native.slack.com/archives/C01N6P7KR6W/p1764961932853649.

I would like to be able to convert some boolean like strings to booleans. Something that improves the following

```yaml
transform/string_to_boolean:
  log_statements:
    - statements:
      - set(log.attributes["my.attribute"], true) where ToLowerCase(log.attributes["my.attribute"]) == "true"
      - set(log.attributes["my.attribute"], false) where ToLowerCase(log.attributes["my.attribute"]) == "false"
```

### Describe the solution you'd like

A function such as `ParseBool` which takes inspiration from [ParseInt](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/pkg/ottl/ottlfuncs#parseint). It could work on non string objects as well. The syntax could look like

```yaml
transform/boolean:
  log_statements:
    - statements:
      - set(log.attributes(["my.attribute"]), ParseBool(log.attributes["my.attribute"])
```

for my string specific case I would expect something like

```yaml
transform/boolean:
  log_statements:
    - statements:
      - set(log.attributes(["my.attribute"]), ParseBool(log.attributes["my.attribute"]) where IsString(log.attributes["my.attribute"])
```

### Describe alternatives you've considered

_No response_

### Additional context

If this is something the maintainers feel like supporting I can contribute a PR

### Tip

<sub>[React](https://github.blog/news-insights/product-news/add-reactions-to-pull-requests-issues-and-comments/) with 👍 to help prioritize this issue. Please use comments to provide useful context, avoiding `+1` or `me too`, to help us triage it. Learn more [here](https://opentelemetry.io/community/end-user/issue-participation/).</sub>

### PR DESCRIPTION
**Pull request by danelson**
#### Description
Adds a `Bool` function. This function can be used to convert values to booleans.

```yaml
- set(attributes["my.bool"], Bool("true"))
```

#### Link to tracking issue
Closes https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/44770

#### Testing
Added new unit and e2e tests.

#### Documentation
Added function doc.

### CODE REVIEW

File: pkg/ottl/ottlfuncs/func_bool_test.go [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+			expected: true,
+		},
+		{
+			name:     "string 0",
+			value:    "0",
+			expected: false,
+		},
+		{
+			name:     "empty string",
+			value:    "",
```
Comment (iblancasa) [inline pkg/ottl/ottlfuncs/func_bool_test.go:102]: Maybe this should be `false`? Like in Python

File: pkg/ottl/ottlfuncs/func_bool_test.go [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+			expected: false,
+		},
+		{
+			name:     "nil",
+			value:    nil,
+			expected: nil,
+		},
+		{
+			name:     "some struct",
+			value:    struct{}{},
```
Comment (iblancasa) [inline pkg/ottl/ottlfuncs/func_bool_test.go:149]: Should we also test an empty slice?

File: pkg/ottl/ottlfuncs/func_bool_test.go [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+			value:    "TrUe",
+			expected: nil,
+			err:      true,
+		},
+		{
+			name:     "string FaLsE",
+			value:    "FaLsE",
+			expected: nil,
+			err:      true,
+		},
```
Comment (edmocosta) [inline pkg/ottl/ottlfuncs/func_bool_test.go:63]:
```suggestion
		},
		{
			name:     "string t",
			value:    "t",
			expected: true,
		},
		{
			name:     "string f",
			value:    "f",
			expected: false,
		},
		{
			name:     "string T",
			value:    "T",
			expected: true,
		},
		{
			name:     "string F",
			value:    "F",
			expected: false,
		},
```

Those are also valid string values.

File: pkg/ottl/ottlfuncs/README.md [hunk 4]
```diff
@@ -560,6 +561,46 @@ Examples:
 
 - `Base64Decode(resource.attributes["encoded field"])`
 
+### Bool
+
+`Bool(value)`
+
+The `Bool` Converter converts the `value` to a bool type.
+
+The returned type is bool.
```
Comment (edmocosta) [inline pkg/ottl/ottlfuncs/README.md]:
```suggestion
The returned type is `bool`.
```

File: pkg/ottl/ottlfuncs/README.md [hunk 5]
```diff
@@ -560,6 +561,46 @@ Examples:
...
+
+The input `value` types:
```
Comment (edmocosta) [inline pkg/ottl/ottlfuncs/README.md]:
```suggestion
The accepted input `value` types are:
```

File: pkg/ottl/ottlfuncs/README.md [hunk 6]
```diff
@@ -560,6 +561,46 @@ Examples:
...
+    | `"true"` | `true` |
+    | `"false"` | `false` |
+    | `"True"` | `true` |
+    | `"False"` | `false` |
+    | `"TRUE"` | `true` |
+    | `"FALSE"` | `false` |
+    | `"1"` | `true` |
+    | `"0"` | `false` |
+
+If `value` is another type or parsing failed nil is always returned.
```
Comment (edmocosta) [inline pkg/ottl/ottlfuncs/README.md]:
```suggestion
If `value` is another type or parsing failed, `nil` is always returned.
```

File: pkg/ottl/ottlfuncs/README.md [hunk 7]
```diff
@@ -560,6 +561,46 @@ Examples:
...
+    | Input | Output |
+    | --- | --- |
+    | `"true"` | `true` |
+    | `"false"` | `false` |
+    | `"True"` | `true` |
+    | `"False"` | `false` |
+    | `"TRUE"` | `true` |
+    | `"FALSE"` | `false` |
+    | `"1"` | `true` |
+    | `"0"` | `false` |
```
Comment (edmocosta) [inline pkg/ottl/ottlfuncs/README.md]:
```suggestion
- `bool`: Returns the value without changes.
- `float64`: Returns `true` if non-zero, otherwise `false`.
- `int64`: Returns `true` if non-zero, otherwise `false`.
- `string`: Tries to parse a boolean from the string. It returns `true` for "1", "t", "T", "true", "TRUE", or "True"; returns `false` for "0", "f", "F", "false", "FALSE" or "False".
- `nil`: Returns `nil`.
```

**Review by edmocosta — COMMENTED** [review]
Thank you for working on this @danelson, it looks good, just a few suggestions.

File: pkg/ottl/ottlfuncs/func_bool_test.go (hunk 1 above)
Comment (edmocosta) [reply pkg/ottl/ottlfuncs/func_bool_test.go:102]: That's a great question, I think I'd stick to the `strconv.ParseBool` approach (used by `ottl.BoolLikeGetter`) and continue to treat empty strings as errors instead of `false`. It's also consistent with other type conversions we support such as `Int` and `Double`, that also relies on the `strconv` package for that.

File: pkg/ottl/ottlfuncs/func_bool_test.go (hunk 2 above)
Comment (danelson) [reply pkg/ottl/ottlfuncs/func_bool_test.go:149]: I don't think it hurts, added

**Comment by danelson** [comment]
> Thank you for working on this @danelson, it looks good, just a few suggestions.

No problem and thanks for the feedback. I think I addressed everything. Please let me know if there is anything else

**Review by edmocosta — APPROVED** [review]
